src/camunda_worker/tasks/tasks/celery.py

```diff
@@ -0,0 +1,59 @@
+from django.conf import settings
+
+from celery.utils.log import get_task_logger
+
+from camunda_worker.external_tasks.camunda import fetch_and_lock
+from camunda_worker.external_tasks.constants import Statuses
+from camunda_worker.external_tasks.models import FetchedTask
+
+from ...celery import app
+from ..api import complete, execute
+
+logger = get_task_logger(__name__)
+
+__all__ = ("task_fetch_and_lock", "task_execute_and_complete")
+
+
+@app.task()
+def task_fetch_and_lock():
+    worker_id, num_tasks, tasks = fetch_and_lock(settings.MAX_TASKS)
+
+    logger.info("fetched %r tasks with %r", num_tasks, worker_id)
+
+    for task in tasks:
+        task_execute_and_complete(task.id)
+    return num_tasks
+
+
+@app.task()
+def task_execute_and_complete(fetched_task_id):
+    fetched_task = FetchedTask.objects.get(id=fetched_task_id)
```

Comment:
I'd filter this queryset down to tasks that have a status that qualifies them for processing - so you don't accidentally execute a completed task.

Reply:
the task invokes `execute` and `complete` api, which process different statuses and can raise specified exceptions if the statuses are not correct

Reply:
right, forgot about that :-)